packages/contractkit/src/wrappers/BaseWrapper.ts

```diff
@@ -46,3 +29,88 @@ export function toBigNumber(input: string) {
 export function toNumber(input: string) {
   return parseInt(input, 10)
 }
+
+export function parseNumber(input: string | number | BigNumber) {
+  return new BigNumber(input).toString()
+}
+
+type Parser<A, B> = (input: A) => B
+
+export function tupleParser<A0, B0>(parser0: Parser<A0, B0>): (...args: [A0]) => [B0]
+export function tupleParser<A0, B0, A1, B1>(
+  parser0: Parser<A0, B0>,
+  parser1: Parser<A1, B1>
+): (...args: [A0, A1]) => [B0, B1]
+export function tupleParser<A0, B0, A1, B1, A2, B2>(
+  parser0: Parser<A0, B0>,
+  parser1: Parser<A1, B1>,
+  parser2: Parser<A2, B2>
+): (...args: [A0, A1, A2]) => [B0, B1, B2]
+export function tupleParser<A0, B0, A1, B1, A2, B2, A3, B3>(
+  parser0: Parser<A0, B0>,
+  parser1: Parser<A1, B1>,
+  parser2: Parser<A2, B2>,
+  parser3: Parser<A3, B3>
+): (...args: [A0, A1, A2, A3]) => [B0, B1, B2, B3]
+export function tupleParser(...parsers: Array<Parser<any, any>>) {
+  return (...args: any[]) => zip((parser, input) => parser(input), parsers, args)
+}
+
+type ProxyCallArgs<A extends any[], B extends any[], C, D> =
```

Comment:
Not entirely clear what's happening here. Do you think more verbose type parameter names could help?

Reply:
I can add a comment explaining.
That's actually some type magic. Basically, we want to optionally preparse and/or postparse a call. So, ProxyCallArgs handles all possible combination with the correct typing, and makes the proxyCall() return type always what's to be expected

Reply:
I think it could be helpful to use meaningful type parameter names anyways, rather than `A`, `B`, etc. Easy to get lost with those.